From 92a81ba1e3f7b642cd829cb8e398d733e18b4ab6 Mon Sep 17 00:00:00 2001
From: ningyu <ningyu@eswincomputing.com>
Date: Thu, 21 Aug 2025 20:28:35 +0800
Subject: [PATCH 430/533] WIN2030-19163:fix:combine temp and voltage in
 testreg0

Changelogs:
Because test register1~3 are used by EP mode, so combine temperature and voltage of pvt0 and store in test register0

Change-Id: If1cf700939936f7cea87560010c1b8233f008a8c
Signed-off-by: ningyu <ningyu@eswincomputing.com>
---
 drivers/hwmon/eswin_dummy_pvt.c | 20 ++++++++------------
 drivers/hwmon/eswin_pvt.h       |  4 +---
 2 files changed, 9 insertions(+), 15 deletions(-)

diff --git a/drivers/hwmon/eswin_dummy_pvt.c b/drivers/hwmon/eswin_dummy_pvt.c
index 711f28481427..d4d9c46de910 100644
--- a/drivers/hwmon/eswin_dummy_pvt.c
+++ b/drivers/hwmon/eswin_dummy_pvt.c
@@ -227,24 +227,20 @@ static int eswin_pvt_read_data(struct pvt_hwmon *pvt, enum pvt_sensor_type type,
 
 	offset = pvt->regmap_offset;
 
-	switch (type) {
-		case PVT_TEMP:
-			offset += LPCPU_PVT_TEMP_REG_OFFSET;
-			break;
-		case PVT_VOLT:
-			offset += LPCPU_PVT_VOLT_REG_OFFSET;
-			break;
-		default:
-			return -EINVAL;
-	}
+	if ((type != PVT_TEMP) && (type != PVT_VOLT))
+		return -EINVAL;
 
 	regmap_read(pvt->regmap, offset, &data);
 
+	/****** dummy register **********
+	[16:0] stores the temperature value
+	[31:16] stores the voltage value
+	********************************/
 	if (type == PVT_TEMP)
-		*val = eswin_pvt_calc_poly(&poly_N_to_temp, data);
+		*val = eswin_pvt_calc_poly(&poly_N_to_temp, (data & 0xffff));
 
 	else if (type == PVT_VOLT)
-		*val = eswin_pvt_calc_poly(&poly_N_to_volt, data);
+		*val = eswin_pvt_calc_poly(&poly_N_to_volt, (data >> 16));
 	else
 		*val = data;
 
diff --git a/drivers/hwmon/eswin_pvt.h b/drivers/hwmon/eswin_pvt.h
index fe30569f4897..c0baaf92ab6d 100644
--- a/drivers/hwmon/eswin_pvt.h
+++ b/drivers/hwmon/eswin_pvt.h
@@ -110,9 +110,7 @@
 #define PVT_TRIM_STEP		(PVT_TRIM_TEMP / PVT_TRIM_MAX)
 #define PVT_TRIM_DEF		0
 #define PVT_TOUT_MIN		(NSEC_PER_SEC / 3000)
-# define PVT_TOUT_DEF		0
-#define LPCPU_PVT_TEMP_REG_OFFSET	0x0
-#define LPCPU_PVT_VOLT_REG_OFFSET	0x4
+#define PVT_TOUT_DEF		0
 
 /*
  * enum pvt_sensor_type - ESWIN PVT sensor types (correspond to each PVT
-- 
2.52.0

